core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -517,12 +518,11 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
     }
 
     //FIXME remove after transition period
-    private boolean isThisGraalVMRCObsolete() {
-        final String vmName = System.getProperty("java.vm.name");
-        log.info("Running Quarkus native-image plugin on " + vmName);
+    private boolean isThisGraalVMRCObsolete(String graalvmHome) {
+        log.info("Running Quarkus native-image plugin on " + graalvmHome);
         final List<String> obsoleteGraalVmVersions = Arrays.asList("-rc9", "-rc10", "-rc11", "-rc12", "-rc13", "-rc14",
                 "-rc15", "-rc16", "19.0.0");
-        final boolean vmVersionIsObsolete = obsoleteGraalVmVersions.stream().anyMatch(vmName::contains);
+        final boolean vmVersionIsObsolete = obsoleteGraalVmVersions.stream().anyMatch(graalvmHome::contains);
```

Comment:
Do you expect the path to GraalVM to contain the version number?

Reply:
No, I think it doesn't always contain version number, but it would be better than before.

Reply:
Sorry I disagree. The previous version of this was working good enough for me, while this wouldn't work at all.

Reply:
Okay, Thank you. How about use ```$GRAALVM_HOME/release``` to check instead of ```System.getProperty("java.vm.name");```?

Reply:
By the way, your JAVA_HOME is set to GraalVM?
I set JAVA_HOME to adoptopenjdk-8.jdk.
so if I use ```System.getProperty("java.vm.name");```, ```vmName``` variable is set ```OpenJDK 64-Bit Server VM``` and checking version didn't work well on my mac environment...

Reply:
Yes I do set JAVA_HOME to the same as GRAALVM_HOME .. I'd recommend this as you want them to match. Of course they don't have to match, especially if you're not going to build native-images.

Maybe we should get rid of `GRAALVM_HOME` and require this?

Reply:
Oh... I completely understanded! thank you very much!!
I just set environment variables from this document 
https://quarkus.io/guides/building-native-image-guide.html
```
・JDK 1.8+ installed with JAVA_HOME configured appropriately
・GraalVM installed from the GraalVM web site. Using the community edition is enough. Version 19.0.2 is required.
・The GRAALVM_HOME environment variable configured appropriately
...
```
and I misunderstanded I shoud set JAVA_HOME, GRAALVM_HOME separately.

maybe, these people are misunderstanding like me and missed the checking version, so the error log ```Out of date RC build of GraalVM detected! Please upgrade to GraalVM 19.0.2``` wasn't shown up.
https://github.com/quarkusio/quarkus/issues/3015

Therefore it makes them hard to handle these issue themselves.

Reply:
Thank you for your advice. I fixed to use system property, and it completely worked.